web/admin/composables/useActor.ts

```diff
@@ -0,0 +1,28 @@
+import { Actor } from "../../proto/bff/v1/types_pb";
+
+type ActorWithInfo = Actor & {
+  isAdmin: boolean;
+  isModOrHigher: boolean;
+};
+
+export default async function (): Promise<Ref<ActorWithInfo>> {
+  const actor = useState<ActorWithInfo>("actor");
+
+  if (!actor.value) {
+    const api = await useAPI();
+    const user = await useUser();
+    const result = (await api
+      .getActor({
```

Comment:
Could this use the "public" API's `GetMe` RPC instead - that RPC will always allow a user to introspect themselves whereas this call is dependent on their ability to fetch an actor.

Reply:
Good idea! That didn’t occur to me because the `useAPI` only has the `ModerationService`, so I added a `useUserAPI` composable for the `UserService`.